radis/lbl/base.py

```diff
@@ -2617,16 +2951,30 @@ def _calc_linestrength_noneq(self):
             #            # (tested + see https://stackoverflow.com/questions/44954514/efficient-way-to-conditionally-populate-elements-in-a-pandas-groupby-object-pos)
             #
             # partition function
-            dgb = df.groupby(by=["id", "iso"])
-            for (id, iso), idx in dgb.indices.items():
+            if "id" in df:
+                dgb = df.groupby(by=["id", "iso"])
+                for (id, iso), idx in dgb.indices.items():
+                    molecule = get_molecule(id)
+                    state = self.input.state
+                    parsum = self._get_parsum(molecule, iso, state)
+                    df.at[idx, "Qref"] = parsum.at(Tref, update_populations=False)
```

Comment:
The Qref column should definitly be replaced by a map()

Reply:
Done.

Reply:
Super efficient, congrats ! 👏